frontend/src/context/SocketContext.tsx

```diff
@@ -0,0 +1,42 @@
+import { PropsWithChildren, createContext, useEffect, useState } from "react";
+import { socket } from "./socket";
+import { InitialSocketContext } from "../types/InitialSocketContext";
+
+const intialContext: InitialSocketContext = {
+	loggedInUsers: [],
+	currentChat: "",
+	setChat: () => {},
+};
+
+export const socketContext = createContext(intialContext);
+
+const SocketContextProvider = ({ children }: PropsWithChildren) => {
+	const [loggedInUsers, setLoggedInUsers] = useState<string[]>([]);
+	const [currentChat, setCurrentChat] = useState<string>("");
+
+	useEffect(() => {
+		const addConnectedUser = (username: string) => {
```

Comment:
How should I do this with the set state fn being referenced?

Reply:
You also pass the state setter function as an argument to this function

Reply:
Okay so this is probably the best we can get because TS can be really silly sometimes with native DOM stuff but you will have to do a bit of type casting which isn't ideal but it's better than any. Here is what I was able to get:

![image](https://github.com/johnsonav1992/jbros-chat/assets/97169070/d511c18b-038c-43e2-ab25-84fcd562ce31)